js/web/lib/onnxjs/graph.ts

```diff
@@ -585,33 +585,48 @@ class GraphImpl implements Graph, Graph.Transformer {
   finalizeGraph() {
     let offset = 0;
     // delete all nodes that are not being executed
+    // The graph is represented using these two arrays
+    // this._nodes - Array holding the kernels to execute - each entry is a kernel pointing to this._allData
+    // this._allData - hold 2 fields - to [] & from [] - these arrays hold the graph map for inputs and outputs per node
+    // reducedMap - remapping the graph after reading the flag 'executeNode' 
+    const reducedGraph = new Array (this._nodes.length,0);
```

Comment:
you can use `new Array<number>(...)` in typescript to hint the compiler for type info